ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -220,24 +224,42 @@ func (h *FaultHandler) startNetworkBlackholePort(ctx context.Context, protocol,
 			"taskArn": taskArn,
 		})
 
-		// Appending a new rule based on the protocol and port number from the request body
-		appendRuleCmdString := nsenterPrefix + fmt.Sprintf(iptablesAppendChainRuleCmd, requestTimeoutSeconds, chain, protocol, port)
-		cmdOutput, err = h.runExecCommand(ctx, strings.Split(appendRuleCmdString, " "))
-		if err != nil {
-			logger.Error("Unable to append rule to chain", logger.Fields{
-				"netns":   netNs,
-				"command": appendRuleCmdString,
+		// Helper function to run iptables rule change commands
+		var execRuleChangeCommand = func(cmdString string) (string, error) {
+			// Appending a new rule based on the protocol and port number from the request body
+			cmdOutput, err = h.runExecCommand(ctx, strings.Split(cmdString, " "))
+			if err != nil {
+				logger.Error("Unable to add rule to chain", logger.Fields{
+					"netns":   netNs,
+					"command": cmdString,
+					"output":  string(cmdOutput),
+					"taskArn": taskArn,
+					"error":   err,
+				})
+				return string(cmdOutput), err
+			}
+			logger.Info("Successfully added new rule to iptable chain", logger.Fields{
+				"command": cmdString,
 				"output":  string(cmdOutput),
 				"taskArn": taskArn,
-				"error":   err,
 			})
-			return string(cmdOutput), err
+			return "", nil
+		}
+
+		// Add a rule to accept all traffic to TMDS
+		protectTMDSRuleCmdString := nsenterPrefix + fmt.Sprintf(iptablesAppendChainRuleCmd,
```

Comment:
Nice!

One qq - in this way, we will create two rules. Is that possible to have only one by using `!` to allow TMDS IP and block others? https://unix.stackexchange.com/questions/11851/iptables-allow-certain-ips-and-block-all-other-connection

Reply:
ACCEPT and DROP targets are terminating so the TMDS rule always takes priority over the DROP rule since TMDS one is added first. I guess `!` could be used also but having two rules is straightforward to me as well.

Reply:
Sounds good!